File: SIDC6 - 04 Comunicação Indireta - MQTT/Servidor/Maquinario.py
import paho.mqtt.client as mqtt
import time
import random

broker_address = "localhost"
port = 1883


def on_connect(client, userdata, flags, rc):
    if rc == 0:
        print("Conectado ao broker MQTT.")
        client.publish("temperatura/maquina", payload="70", qos=0, retain=False)
    else:
        print("Falha na conexão com o broker MQTT.")

def on_message(client, userdata, msg):
    if msg.topic == "controle/maquina":
        if msg.payload.decode() == "AUMENTAR":
            temperatura = random.randint(60, 75)
        elif msg.payload.decode() == "DIMINUIR":
            temperatura = random.randint(100, 120)
        client.publish("temperatura/maquina", payload=str(temperatura), qos=0, retain=False)

client = mqtt.Client("MaquinarioMQTT")
client.on_connect = on_connect
client.on_message = on_message
temperatura = 70

client.connect(broker_address, port, 60)

while True:
    temperatura += random.randint(-15, 15)
    client.publish("temperatura/maquina", payload=str(temperatura), qos=0, retain=False)
    time.sleep(random.uniform(0.5, 1))

client.loop_forever()

File: SIDC6 - 04 Comunicação Indireta - MQTT/Implementação JavaScript/Maquinario.js
const mqtt = require('mqtt');

const brokerUrl = 'mqtt://localhost:1883';

const client = mqtt.connect(brokerUrl, { clientId: 'MaquinarioMQTT' });

let temperatura = 70;

client.on('connect', () => {
  console.log('Conectado ao broker MQTT.');
  client.publish('temperatura/maquina', temperatura.toString());
  client.subscribe('controle/maquina');
});

client.on('message', (topic, message) => {
  if (topic === 'controle/maquina') {
    if (message.toString() === 'AUMENTAR') {
      temperatura = Math.min(temperatura + Math.floor(Math.random() * 16), 100);
			console.log(`Comando: AUMENTAR; Temperatura: ${temperatura}`);
    } else if (message.toString() === 'DIMINUIR') {
      temperatura = Math.max(temperatura - Math.floor(Math.random() * 16), 0);
			console.log(`Comando: DIMINUIR; Temperatura: ${temperatura}`);
    }
    client.publish('temperatura/maquina', temperatura.toString());
  }
});

setInterval(() => {
  temperatura += Math.floor(Math.random() * 31) - 15;
  client.publish('temperatura/maquina', temperatura.toString());
}, Math.random() * 500 + 500);
